### ISSUE TITLE
Fix CLI documentation

### ISSUE THREAD

**Issue by chmreid** [DataBiosphere/data-store-cli#50]
Part of #49

Sphinx documentation is currently throwing lots of errors and warnings when trying to build.

Many of the "Explicit markup ends without a blank line" warnings (e.g., on endpoints `get_bundle`, `get_bundles_all`, etc.) are due to the markup in `dss-api.yml`.

Same is true of many of the "Unexpected section title" warnings, those are caused by section headers in `dss-api.yml` endpoint descriptions.

Don't know why this markup is a problem here but not for HumanCellAtlas/dcp-cli

```
$ make html
Running Sphinx v2.4.1
making output directory... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 9 source files that are out of date
updating environment: [new config] 9 added, 0 changed, 0 removed
reading sources... [100%] tutorials/index
/Users/charles/codes/dbio/cli/dbio/dss/__init__.py:docstring of dbio.dss.DSSClient:4: WARNING: Unexpected section title.

HTTP semantics
##############
/Users/charles/codes/dbio/cli/dbio/dss/__init__.py:docstring of dbio.dss.DSSClient:31: WARNING: Unexpected section title.

Subscriptions
#############
/Users/charles/codes/dbio/cli/dbio/dss/__init__.py:docstring of dbio.dss.DSSClient:57: WARNING: Unexpected section title.

Special String Formats
######################
/Users/charles/codes/dbio/cli/dbio/dss/__init__.py:docstring of dbio.dss.DSSClient:62: WARNING: Unexpected section title.

Pagination
##########
/Users/charles/codes/dbio/cli/dbio/dss/__init__.py:docstring of dbio.dss.DSSClient.get_bundle:16: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/charles/codes/dbio/cli/dbio/dss/__init__.py:docstring of dbio.dss.DSSClient.get_bundles_all:16: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/charles/codes/dbio/cli/dbio/dss/__init__.py:docstring of dbio.dss.DSSClient.get_collections:16: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/charles/codes/dbio/cli/dbio/dss/__init__.py:docstring of dbio.dss.DSSClient.get_events:16: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/charles/codes/dbio/cli/dbio/dss/__init__.py:docstring of dbio.dss.DSSClient.get_file:21: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/charles/codes/dbio/cli/dbio/dss/__init__.py:docstring of dbio.dss.DSSClient.post_search:17: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/charles/codes/dbio/cli/dbio/dss/__init__.py:docstring of dbio.dss.DSSClient.post_search:33: WARNING: Unexpected section title.

Index design
############
/Users/charles/codes/dbio/cli/dbio/dss/__init__.py:docstring of dbio.dss.DSSClient.post_search:76: WARNING: Unexpected section title.

Limitations to index design
###########################
WARNING: autodoc: failed to import method 'AuthClient.get_jwks.json' from module 'dbio.auth'; the following exception was raised:
Traceback (most recent call last):
  File "/Users/charles/codes/dbio/cli/vp/lib/python3.6/site-packages/sphinx/util/inspect.py", line 238, in safe_getattr
    return getattr(obj, name, *defargs)
AttributeError: type object 'AuthClient' has no attribute 'get_jwks'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/charles/codes/dbio/cli/vp/lib/python3.6/site-packages/sphinx/ext/autodoc/importer.py", line 71, in import_object
    obj = attrgetter(obj, attrname)
  File "/Users/charles/codes/dbio/cli/vp/lib/python3.6/site-packages/sphinx/ext/autodoc/__init__.py", line 226, in get_attr
    return autodoc_attrgetter(self.env.app, obj, name, *defargs)
  File "/Users/charles/codes/dbio/cli/vp/lib/python3.6/site-packages/sphinx/ext/autodoc/__init__.py", line 1575, in autodoc_attrgetter
    return safe_getattr(obj, name, *defargs)
  File "/Users/charles/codes/dbio/cli/vp/lib/python3.6/site-packages/sphinx/util/inspect.py", line 254, in safe_getattr
    raise AttributeError(name)
AttributeError: get_jwks

/Users/charles/codes/dbio/cli/dbio/auth/__init__.py:docstring of dbio.auth.AuthClient:8: WARNING: Unexpected section title.

Pagination
##########
/Users/charles/codes/dbio/cli/dbio/auth/__init__.py:docstring of dbio.auth.AuthClient.get_oauth_authorize:21: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/charles/codes/dbio/cli/dbio/auth/__init__.py:docstring of dbio.auth.AuthClient.get_v1_group_roles:16: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/charles/codes/dbio/cli/dbio/auth/__init__.py:docstring of dbio.auth.AuthClient.get_v1_group_users:16: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/charles/codes/dbio/cli/dbio/auth/__init__.py:docstring of dbio.auth.AuthClient.get_v1_groups:16: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/charles/codes/dbio/cli/dbio/auth/__init__.py:docstring of dbio.auth.AuthClient.get_v1_resource:16: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/charles/codes/dbio/cli/dbio/auth/__init__.py:docstring of dbio.auth.AuthClient.get_v1_resource_id_members:16: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/charles/codes/dbio/cli/dbio/auth/__init__.py:docstring of dbio.auth.AuthClient.get_v1_resource_ids:16: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/charles/codes/dbio/cli/dbio/auth/__init__.py:docstring of dbio.auth.AuthClient.get_v1_resource_policies:16: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/charles/codes/dbio/cli/dbio/auth/__init__.py:docstring of dbio.auth.AuthClient.get_v1_resources:16: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/charles/codes/dbio/cli/dbio/auth/__init__.py:docstring of dbio.auth.AuthClient.get_v1_roles:16: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/charles/codes/dbio/cli/dbio/auth/__init__.py:docstring of dbio.auth.AuthClient.get_v1_user_groups:16: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/charles/codes/dbio/cli/dbio/auth/__init__.py:docstring of dbio.auth.AuthClient.get_v1_user_owns:16: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/charles/codes/dbio/cli/dbio/auth/__init__.py:docstring of dbio.auth.AuthClient.get_v1_user_roles:16: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/Users/charles/codes/dbio/cli/dbio/auth/__init__.py:docstring of dbio.auth.AuthClient.get_v1_users:16: WARNING: Explicit markup ends without a blank line; unexpected unindent.
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] tutorials/index
generating indices...  genindex py-modindexdone
highlighting module code... [100%] dbio.util
writing additional pages...  searchdone
copying static files... ... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 28 warnings.

The HTML pages are in _build/html.
```

**Comment by chmreid** [comment]
## Debugging information

The data-store-cli parses the data store swagger file and turns various endpoints into methods on-the-fly.

To update/modify the CLI tool, we need to update/modify the swagger file `dss-api.yml` in the DataBiosphere/data-store repo. This means we either need to repeatedly re-deploy the data store (bad), or run a local API server that hosts a local swagger so we can cut the deployment step out of the loop (good).

Procedure for running a local API server is described below.

## Debugging procedure

Clone a copy of DataBiosphere/data-store, and source the `environment` file in that repo.

Run the local API server by running the script `./dss-api`. This runs a local API server at `http://localhost:5000`. The new swagger file is available at `http://localhost:5000/v1/swagger.json`

Update the data-store-cli config file to point to the localhost URL:

```
$ cat ~/.config/dbio/config.json | jq
{
  "log_level": "INFO",
  "client_id": "foo",
  "DSSClient": {
    "swagger_url": "http://localhost:5000/v1/swagger.json"
  },
  "AuthClient": {
    "swagger_url": "https://auth.ucsc.ucsc-cgp-redwood.org/swagger.json"
  }
}
```

## Refreshing swagger

The CLI tool caches the swagger file, so when you update your swagger file and restart the local API server, you will also need to clear the CLI tool's local swagger cache:

```
dbio clear-dbio-cache
```

**Comment by chmreid** [comment]
Closed by databiosphere/data-store#98
